src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,54 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task that has to be completed before a certain date.
+ */
+public class Deadline extends Task implements Timeable {
+    protected LocalDate by;
```

Comment:
Might want to change this to private instead of protected. For java, unlike C++ where protected makes the variable inaccessible to other classes besides the derived classes, in java, as long as a class is in the same package, they can access the protected variable.

If that is not your intention, you may want to change to private since Deadline class is only using this variable (correct me if I'm wrong).